Sources/tinys3/model/AWSProfileConfigFileParser.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+
+/// Parser for ~/.aws/config and ~/.aws/credentials files
+struct AWSProfileConfigFileParser {
+    enum FileType {
+        case config
+        case credentials
+    }
+
+    /// Parses the `~/.aws/config` file for profiles and their associated configuration values
+    /// - Returns: A dictionary containing an `AWSProfileConfig` for each profile name found
+    static func profilesFromConfigUserFile() throws -> [AWSProfile: AWSProfileConfig] {
+        let url = FileManager.default.homeDirectoryForCurrentUser
+            .appendingPathComponent(".aws")
+            .appendingPathComponent("config")
+        return try profiles(from: url, fileType: .config)
+    }
+
+    /// Parses the `~/.aws/credentials` file for profiles and their associated configuration values
+    /// - Returns: A dictionary containing an `AWSProfileConfig` for each profile name found
+    static func profilesFromCredentialsUserFile() throws -> [AWSProfile: AWSProfileConfig] {
+        let url = FileManager.default.homeDirectoryForCurrentUser
+            .appendingPathComponent(".aws")
+            .appendingPathComponent("credentials")
+        return try profiles(from: url, fileType: .credentials)
+    }
+
+    /// Parses the profiles and their associated configuration values from a given file
+    /// - Parameters:
+    ///   - path: The URL of the file to parse
+    ///   - isCredentialsFile: Indicates if the file to parse is a credentials file (true) or a config file (false)
+    /// - Returns: A dictionary containing an `AWSProfileConfig` for each profile name found
+    static func profiles(from path: URL, fileType: FileType) throws -> [AWSProfile: AWSProfileConfig] {
+        try profiles(from: String(contentsOf: path), fileType: fileType)
+    }
+
+    /// Parses the profiles and their associated configuration values from a given string
+    ///
+    /// See [the AWS Documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
+    ///
+    /// - Parameters:
+    ///   - string: The content of the file to parse
+    ///   - isCredentialsFile: Indicates if the file to parse is a credentials file (true) or a config file (false)
```

Comment:
```suggestion
    ///   - fileType: Indicates if the file to parse is a credentials file (.credentials) or a config file (.config)
```

Reply:
Nice catch, thanks! Fixed in [018c403](https://github.com/Automattic/hostmgr/pull/93/commits/018c403ad26c6d72ce8a97ddbe03bbddf34dc3de)